### PR TITLE
Introduce ban API

### PR DESCRIPTION
**Pull request by noffle**
Adds the `ban` function, docs, and updates the moderation tests.

### CODE REVIEW

**Comment by cblgh** [comment]
@noffle is this ready to be merged? :)

**Comment by noffle** [comment]
@cblgh yes, if there's no feedback / issues!

File: README.md [hunk 1]
```diff
@@ -123,6 +123,15 @@ Determine whether a user identified by `ip` and/or `key` is banned on `channel`
 or cabal-wide as `cb(err, banned)` for a boolean `banned`. If `channel` is
 omitted, only check cabal-wide.
 
+### cabal.ban(key, [opts], [cb])
```
Comment (okdistribute) [inline README.md:126]: should this be mute/hide/or whatever to mirror commands?

**Comment by noffle** [comment]
Closing in favour of #80.

**Pull request closed without merging**